Src/Coravel/Cache/InMemoryCache.cs

```diff
@@ -22,6 +22,35 @@ public InMemoryCache(IMemoryCache cache)
             this._keys = new HashSet<string>();
         }
 
+        public TResult Get<TResult>(string key)
+        {
+            return this.Get(key, default(TResult));
+        }
+
+        public TResult Get<TResult>(string key, TResult @default)
+        {
+            if (!this._cache.TryGetValue(key, out var value)) return @default;
+
+            if (value is TResult result) return result;
+
+            return @default;
+        }
+
+        public async Task<TResult> GetAsync<TResult>(string key)
+        {
+            return await this.GetAsync(key, default(TResult));
+        }
+
+        public async Task<TResult> GetAsync<TResult>(string key, TResult @default)
+        {
+            return this.Get(key, @default);
```

Comment:
I don't think `GetAsync` is needed. The implementation here isn't even async - so it's misleading to the caller.

`IMemoryCache` has a `GetOrCreateAsync` but the async only applies to the `Create` portion. Accessing the cache is always synchronous.

Unless I'm missing something, I would take it out.

Reply:
![image](https://user-images.githubusercontent.com/7213174/47579706-1d181380-d923-11e8-954c-5b325da27152.png)

VS Code should warn you that it's not async

Reply:
I included this in case other implementations need it on the interface, specifically I have a DB cache that I have written that would need this.

Reply:
Ok that makes sense now. The same would apply to redis etc. too if someone wanted to use that under the covers.

Just remove the async keyword from `GetAsync`. And the changes for the tests should still apply - awesome work! Thanks @evanrobertson 👍